server/survey/collectImport/jobs/surveySchemaReaderJob.js

```diff
@@ -0,0 +1,38 @@
+const parser = require('fast-xml-parser')
+const FileZipUtils = require('../../../../common/file/fileZipUtils')
+
+const Job = require('../../../job/job')
+
+const idmlXmlFileName = 'idml.xml'
+
+/**
+ * Reads the schema from a Collect backup file and saves it into the job context under idmlSource property
+ */
+class SurveySchemaReaderJob extends Job {
+
+  constructor (params) {
+    super('SurveySchemaReaderJob', params)
+  }
+
+  async execute (tx) {
+
+    const zipFile = await FileZipUtils.openFile(this.params.filePath)
+
+    const idmlXml = await FileZipUtils.getEntryAsText(zipFile, idmlXmlFileName)
+
+    const options = {
+      attrNodeName: '_attr',
+      attributeNamePrefix: '',
+      textNodeName: '_text',
+      ignoreAttributes: false,
+      format: false,
+      indentBy: '  ',
+    }
+    const idmlTraversalObj = parser.getTraversalObj(idmlXml, options)
+    const idmlJsonObj = parser.convertToJson(idmlTraversalObj, options)
```

Comment:
I would add all this parsing code to the object collectIdmlParser

Reply:
it doesn't do too much, it's would be just a generic method that parses XML into JSON... it's not related to the Collect IDML

Reply:
yes i know, it was more about to isolate this logic and all dependencies into the collectIdmlParser (which could be moved at this level in the directory?)